optaplanner-core/src/main/java/org/optaplanner/core/impl/domain/variable/inverserelation/ExternalizedSingletonListInverseVariableSupply.java

```diff
@@ -27,7 +27,7 @@
 import org.optaplanner.core.impl.domain.variable.listener.SourcedVariableListener;
 
 /**
- * Alternative to {@link SingletonInverseVariableListener}.
+ * Alternative to {@link SingletonListInverseVariableListener}.
  */
 public class ExternalizedSingletonListInverseVariableSupply<Solution_>
```

Comment:
Are we sure that "...SingletonList..." is a good name?

Reply:
The existing classes for the basic variable are `SingletonInverseVariableListener` and `ExternalizedSingletonInverseVariableSupply`. I assume that adding `List` somewhere is a good way to differentiate the corresponding pair of classes designed for the list variable. But I agree that gluing five nouns together leaves a space for many interpretations.

In both cases they are a supply/listener that provide an "inverse relation" "(shadow) variable" that is a "singleton". So probably we should keep "SingletonInverseVariable" together and just prefix it with "List", so:

- `SingletonInverseVariableListener`
- `ExternalizedSingletonInverseVariableSupply`
- `ListSingletonInverseVariableListener`
- `ExternalizedListSingletonInverseVariableSupply`

Wdyt?

We could also prefix the existing classes with "Basic" but we'd also have to apply that to the "Collection" variant, so:

- `BasicCollectionInverseVariableListener`
- `ExternalizedBasicCollectionInverseVariableSupply`
- `BasicSingletonInverseVariableListener`
- `ExternalizedBasicSingletonInverseVariableSupply`
- `ListSingletonInverseVariableListener`
- `ExternalizedListSingletonInverseVariableSupply`

Reply:
Not a blocker for this PR. Let's continue the naming discussion in the design document.